convert_zdf.py

```diff
@@ -0,0 +1,151 @@
+    )
+    parser.add_argument("--ply", action="store_true", help="Convert to PLY")
+    parser.add_argument("--csv", action="store_true", help="Convert to CSV")
+    parser.add_argument("--txt", action="store_true", help="Convert to text")
+    parser.add_argument(
+        "--png",
+        action="store_true",
+        help="Convert to Portable Network Graphics (3D->2D)",
+    )
+    parser.add_argument("--jpg", action="store_true", help="Convert to JPEG (3D->2D)")
+    parser.add_argument(
+        "--bmp", action="store_true", help="Convert to Windows bitmap (3D->2D)"
+    )
+    parser.add_argument("--tiff", action="store_true", help="Convert to TIFF (3D->2D)")
+
+    return parser.parse_args()
+
+
+def _flatten_point_cloud(point_cloud):
+    """Convert from point cloud to flattened point cloud (with numpy).
+
+    Args:
+        point_cloud: a numpy array
+
+    Returns:
+        2D numpy array, with 7 columns and npixels rows
+
+    """
+    # Convert to numpy 3D array
+    point_cloud = np.dstack(
+        [
+            point_cloud["x"],
+            point_cloud["y"],
+            point_cloud["z"],
+            point_cloud["r"],
+            point_cloud["g"],
+            point_cloud["b"],
+            point_cloud["contrast"],
+        ]
+    )
+    # Flattening the point cloud
+    flattened_point_cloud = point_cloud.reshape(-1, 7)
+    # Just the points without color and contrast
+    # point_cloud = np.dstack([point_cloud['x'],point_cloud['y'],point_cloud['z']])
+    # flattened_point_cloud = point_cloud.reshape(-1,3)
+
+    # Removing nans
+    return flattened_point_cloud[~np.isnan(flattened_point_cloud[:, 0]), :]
+
+
+def _convert_2_ply(frame, file_name: str):
+    """Convert from frame to ply.
+
+    Args:
+        frame: A frame captured by a Zivid camera.
+        file_name: File name without extension
+
+    """
+    print(f"Saving the frame to {file_name}.ply")
+    frame.save(file_name + ".ply")
+
+
+def _convert_2_csv(point_cloud, file_name: str):
+    """Convert from point cloud to csv or txt.
+
+    Args:
+        point_cloud: a numpy array
+        file_name: File name with extension
+
+    """
+    print(f"Saving the frame to {file_name}")
+    np.savetxt(file_name, _flatten_point_cloud(point_cloud), delimiter=",", fmt="%.3f")
+
+
+def _convert_2_2d(point_cloud, file_name: str):
+    """Convert from point cloud to 2D image.
+
+    Args:
+        point_cloud: a numpy array
+        file_name: File name without extension
+
+    """
+    print(f"Saving the frame to {file_name}")
+    rgb = np.dstack([point_cloud["b"], point_cloud["g"], point_cloud["r"]])
+    cv2.imwrite(f"{file_name}", rgb)
+
+
+def _main():
+    options = _options()
+
+    app = zivid.Application()
+
+    for filename_zdf in options.files:
+        file_path = Path(filename_zdf)
+        if file_path.is_file():
+            print(f"Reading {filename_zdf} point cloud")
+            frame = zivid.Frame(filename_zdf)
+            # Getting the point cloud
```

Comment:
I also think this is redundant

Reply:
as in: should be put in each conversion function?